<html>
<head>
<meta http-equiv="Content-Type" content="text/html;charset=utf-8">
<!-- <link type="text/css" rel="stylesheet" href="style.css" /> -->
<!-- <style type="text/css"> -->
<meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <!-- <title>Bootstrap 101 Template</title> -->
    <!-- Bootstrap -->
    <link href="bootstrap/dist/css/bootstrap.min.css" rel="stylesheet">
    <!-- Template borrowed and altered from here: http://getbootstrap.com/examples/starter-template/ -->
    <link href="css/homepage-template.css" rel="stylesheet">
</style>
</head>
	<body>
	<!-- Navbar code borrowed and altered from here: http://getbootstrap.com/examples/starter-template/ -->
	<div class="navbar navbar-inverse navbar-fixed-top" role="navigation">
      <div class="container">
        <div class="navbar-header">
          <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
          <a class="navbar-brand" href="#">Genomics Force Demo</a>
        </div>
        <div class="collapse navbar-collapse">
          <ul class="nav navbar-nav">
            <li class="active"><a href="#">Home</a></li>
            <li><a href="ForceTree.html?FileToOpen=ismeJRDP_CaseControl.json">Live Demo</a></li>
            <li><a href="#contact">Contact</a></li>
          </ul>
        </div><!--/.nav-collapse -->
      </div>
    </div>
		<h1>Draggable force tree for genomics data (Chrome browsers only).</h1>
		<!--<br><a href=ForceTree.html?FileToOpen=testOperon.json>force tree with  rna-seq data</a>-->
		<!--<br><a href=ForceTree.html?FileToOpen=aTree2.json>force tree with gastro 16S data</a>-->
		<!-- YouTube Demo -->
		<br><br><iframe width="420" height="315" src="//www.youtube.com/embed/Sm_fiwpmr0U?rel=0" frameborder="0" allowfullscreen></iframe>
		<br>
		<!-- Link to Demo -->
		<br><a href=ForceTree.html?FileToOpen=ismeJRDP_CaseControl.json>force tree with ISME-J 16S data</a>
		<div id="intro"><h3>Introduction</h3>
		<p>Genomics and bioinformatics data often are generated at many hierarchical levels.  
		For example, an <a href="http://rdp.cme.msu.edu/">RPD</a> 11classification of a 16S rRNA experiment yields 
		calls to phylum, class, order, family and then genus with each level nested within the one above it.  
		So a phylum can have many classes, which in turn can have many orders and so forth.  
		For an rna-seq experiment on bacteria, reads can be mapped to genes, which in turn can be grouped into 
		operons, with one to many genes per operon.  In performing inference, it is not trivial to interpret these 
		multiple levels.  Typically, levels are chosen ahead of time and infernence is constrained to that level.  
		For example, it is common to perform RDP analyses at the genus or phyla level.  
		But such a priori filtering of the data may miss important patterns; for example, for some taxa 
		a difference at the genus level might be important, while for other taxa in the same experiment, a 
		meaningful signal wonâ€™t be observed unless a less derived phylogenetic level is considered.  
		</div>
		<div id="details">
		<h3>Visualizing inference in a way that is hierarchically sensitive to taxonomy</h3>

		<p>In order to visualize such cases, this web page contains a demonstration project for visualization of 
		phylogenetic trees.  In this <a href=ForceTree.html?FileToOpen=ismeJRDP_CaseControl.json> example</a> 
		we consider RDP classifications of a <a href=http://www.nature.com/ismej/journal/v6/n10/full/ismej201243a.html>
		previously published experiment</a>.  
		This was a 454-based 16S rRNA experiment in which punch biopsies were collected from <a href=http://www.nature.com/ismej/journal/v6/n10/fig_tab/ismej201243t1.html#figure-title>33 subjects with 
		adenomas and 38 (control) subjects without adenomas.</a>.  In the analysis presented on this web page,
		578,096 sequences were run through the stand-alone version of the RDP classifier (version 2.6).
		Classifications to each phylogenetic level (phylum, class, order, family, genus) were taken if
		the bootstrap value was >= 50. For each taxa at each level in the tree, a t-test was performed
		testing the null hypothesis that the log-normalized distribution of that taxa was the same in 
		case vs. control.
		
		The visualization <a href=ForceTree.html?FileToOpen=ismeJRDP_CaseControl.json>shown here</a> (for Chrome 
		browsers only) shows the phylogenetic tree that results from this analysis.  By default, the size of
		each circle is the number of sequences assigned to each node:
		<br>
		<center><img id="anImage1"  
		 height=400 width=432 align=middle src="treeShowingPhylogeneticLevels.jpg"></img></center><br>
		
		<p>If in the <a href=ForceTree.html?FileToOpen=ismeJRDP_CaseControl.json>demonstration</a> we hover the mouse over each of 
		the large dots at the phylum level, we see that the three most abundant phyla in our samples were Phylum Firmicutes (with 
		340,933 sequences), Phylum Bacteroidetes (with 142,231 sequences) and Phylum Proteobacteria (with 67,926 sequences) as
		we might expect from a human gut micro-environment.
		
		<p>One of the nice things about the demonstration is that we can switch between a traditional circular 
		phylogenetic tree (as in the figure above) and a Force Tree view.  We can do this by clicking on
		the "reanimate" button in the left hand control panel.   On hitting this button, we switch to a view 
		 in which opposing forces position each node.  These forces include "gravity" which tends to pull each node
		to the center of the screen, a repulsive focce between each node and an elastic spring-like force that 
		in this case reflects the phyologentic distance between different taxonomic levels (so for example the
		distance between genus and family will be shorter than the distance between genus and order):
		<br>
		<center><img id="anImage1"  
		 height=400 width=400 align=middle src="treeShowingForceView.jpg"></img></center><br>
		
		<p>
		(We can adjust the gravity force by using the "Gravity adjust" slide bar under the "plot" menu in the demonstration).
		
		<p> To switch back to the constrained view, hit "T" on the keyboard (or click on the "arrange (entire tree)" button
		 in the left control panel).
		
		<p>Another nice thing about the view is that we can drag individual sections of the tree to different regions of the screen.
		In the below view, we have dragged the node representing Proteobacteria to the corner of the screen. 
		We then hit the "A" button (or click on the "arrange (around last selected)" button in the left control panel):
		<br>
		<center><img id="anImage1"  
		 height=400 width=650 align=middle src="proteobacteriaSeparate.jpg"></img></center><br>
		
		<p>This allows us to break out sub-taxa that we are interesed in for a more detailed inspection than is possible
		when the taxa is constrained by all of the other taxa in the tree.<br>
		(Hitting the "V" button repeatedly allows us to toggle between only showing the last selected).
		
		<p>We can use the interface to continue to break down our phylogenetic tree to any desired level of 
		 phylogenetic resolution. For example here we have created a third level
		for Gammaproteobacteria (by dragging the "Gammaproteobacteria" node to where we want it and again hitting "A")
		<br>
		<center><img id="anImage1"  
		 height=400 width=650 align=middle src="threeLevels.jpg"></img></center><br>
		
		<p> Having arranged the view in this way, we can switch to an inference driven view.  By selecting "-log10TTestP"
		under size on the left control panel, we change the view so that the size of each node no longer 
		reflect the number of sequences in each node, but rather the -log10(pValue) where pValue is the result of a t-test
		evaluating the null hypothesis that the distribution of each node is the same in case and control...
		<br>
		<center><img id="anImage1"  
		 height=400 width=650 align=middle src="threeLevelsInfernce.jpg"></img></center><br>
		
		<p> This allows us to visualize inference information in a way that is hierarchcially sensitive to 
		taxonomic information.  Larger nodes are taxa that are more significantly different between case and control.
		   This is a rich visualization that will reveal patterns that may not be obvious when
		 considering taxomomic and inferential relationships separately.  For example, order 
		 Pasteurellales (inidcated by the arrow above) is more significantly different between case
		 and control than any mord derived taxa under Pasteurellales.  By contrast, genus Sphingobium 
		 (also indicated by an arrow ) is more significant than any of its parent taxa higher in the tree.
		 
		 <br><br>(We can view each branch of the tree separate from all other branches by toggling the v key)
		 </div>
		 
		 <!--
		 <h3>Selecting subsets of multi-dimensional datasets in a hierarchically aware matter</h3>
		 <p> Although (obviously) still under development, the demonstration has some rudimentary ability to 
		 	visualize data in a 2-dimensional scatter plot space and relate that visualization to the hierarchical 
		 	space.  In the following, we show the relationship between the -logPValue (for the null hypothesis that 
		 	that the distribution of each taxa is the same in case and control) and the fraction of sequences for each
		 	taxa that were assigned to case:
		 	-->

		 <div id="contact"><h3>Contact Info</h3></div>
		 <p>Todo: insert contact info here.
		
	</body>
	<script type="text/javascript" src="bootstrap/dist/js/bootstrap.min.js"></script>
</html>
